### PR TITLE
Feat/climbing prerequisits

### PR DESCRIPTION
**Pull request by thestonefox**

### CODE REVIEW

File: Scripts/Data/Collection/GameObjectEventStack.cs [hunk 1]
```diff
@@ -144,5 +152,13 @@ public virtual void PopAt(GameObject element)
                 elementEvents[EventIndex - 1].Restored?.Invoke(Stack[EventIndex - 1]);
             }
         }
+
+        /// <summary>
+        /// Aborts the current stack pop process to prevent any further elements from being popped off the stack.
+        /// </summary>
+        public virtual void AbortPop()
```
[the remark this answers is not in the document]
Comment (thestonefox) [reply Scripts/Data/Collection/GameObjectEventStack.cs]: You shouldn't be allowed to set it to false, it's an instant thing that once you set it and the pop loop happens it becomes true again.

If someone randomly sets it to false then wants to set it to true before the pop loop happens then they're doing it wrong :p

File: Prefabs/Interactions/Interactors/SharedResources/Scripts/InteractorExtractor.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿namespace VRTK.Core.Prefabs.Interactions.Interactors
+{
+    using UnityEngine;
+    using VRTK.Core.Event;
+
+    /// <summary>
+    /// Extracts the relevant <see cref="GameObject"/> from an <see cref="InteractorFacade"/>.
```
[the remark this answers is not in the document]
Comment (thestonefox) [reply Prefabs/Interactions/Interactors/SharedResources/Scripts/InteractorExtractor.cs]: I don't mean GameObject to be plural.

You can only extract one GameObject

File: Scripts/Data/Collection/GameObjectEventStack.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (bddckr) [reply Scripts/Data/Collection/GameObjectEventStack.cs]: OK 👍

File: Prefabs/Interactions/Interactors/SharedResources/Scripts/InteractorExtractor.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bddckr) [reply Prefabs/Interactions/Interactors/SharedResources/Scripts/InteractorExtractor.cs]: This class offers two methods that each extract one GameObject.

**Pull request merged**